crates/transaction-emitter-lib/src/transaction_generator/nft_mint.rs

```diff
@@ -79,6 +79,26 @@ pub async fn initialize_nft_collection(
     collection_name: &[u8],
     token_name: &[u8],
 ) {
+    // resync root account sequence number
+    match rest_client.get_account(root_account.address()).await {
```

Comment:
nit: May be define a separate function `resync_root_account_if_needed` for this?

Reply:
I actually added swarm.chain_info().resync_root_account_seq_num but we don't have that here. maybe we need resync_account function, but it is not clear where to put those utility functions.

there is a bigger cleanup needed for all the utility functions between chain_info, aptos_public_info, utilities for local swarm etc, but I didn't get chance to get to it

I'll leave this separately